fastify-opentelemetry.d.ts

```diff
@@ -0,0 +1,32 @@
+import { FastifyPluginCallback, FastifyReply, FastifyRequest } from 'fastify';
+import { context, getSpan, propagation, TextMapGetter, TextMapSetter, Tracer } from "@opentelemetry/api"
+
+declare module 'fastify' {
+  interface FastifyRequest {
+    activeSpan: () => ReturnType<typeof getSpan>,
+    context: () => ReturnType<typeof context.active>,
+    tracer: () => Tracer,
+    inject: <Carrier> (carrier: Carrier, setter?: TextMapSetter) => ReturnType<typeof propagation.inject>,
+    extract: <Carrier> (carrier: Carrier, getter?: TextMapGetter) => ReturnType<typeof propagation.extract>,
+  }
+}
+
+/**
+ * Options for the OpenTelemetry plugin.
+ */
+export interface OpenTelemetryPluginOptions {
+  serviceName?: string,
+  exposeApi?: boolean,
+  formatSpanName?: (serviceName: string, raw: FastifyRequest) => string,
```

Comment:
The `raw` param isn't the entire `FastifyRequest`, it's just the `request.raw` property. Using node's `Http2ServerRequest` interface should work.

Reply:
Will switch out to use:
```
formatSpanName?: (serviceName: string, raw: FastifyRequest['raw']) => string,
```